pkg/reconciler/source/config_watcher.go

```diff
@@ -20,143 +20,232 @@ import (
 	"context"
 
 	"go.uber.org/zap"
+
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"knative.dev/eventing/pkg/logging"
 	"knative.dev/pkg/configmap"
-	pkgLogging "knative.dev/pkg/logging"
+	pkglogging "knative.dev/pkg/logging"
 	"knative.dev/pkg/metrics"
 	tracingconfig "knative.dev/pkg/tracing/config"
+)
 
-	"knative.dev/eventing/pkg/logging"
+const (
+	EnvLoggingCfg = "K_LOGGING_CONFIG"
+	EnvMetricsCfg = "K_METRICS_CONFIG"
+	EnvTracingCfg = "K_TRACING_CONFIG"
 )
 
+type EnvVarsGenerator interface {
+	ToEnvVars() []corev1.EnvVar
```

Comment:
Good idea, wouldn't hurt to have them. I was actually wondering over the weekend if we could avoid the expensive-ish serialization on every call by storing per-serialized configs instead of structs, but that's a longer discussion.

Reply:
I don't think it's a good idea, because this serialization is cheap, doesn't affect so much the controllers which already have other heavy work, and there could be a use case where user needs that controller configuration materialized

Reply:
Fair enough, will do.

Reply:
Done in e640e85